test/integration/server-selection/server_selection.test.ts

```diff
@@ -0,0 +1,248 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { Collection, MongoClient } from '../../../src';
+import { ConnectionPool } from '../../../src/cmap/connection_pool';
+
+const TEST_METADATA: MongoDBMetadataUI = {
```

Comment:
Note that I didn't add specific tests to ensure that we actually are incrementing the operation count.  The only way I could think to test this would be

1. set a failpoint with a really long delay on the server
2. don't await the command when executing it
3. assert that the count is incremented
4. await the comment now
5. assert the count is zero again

This felt really brittle.  Definitely open to suggestions on better ways to test this.

Ultimately, I decided against writing tests for this scenario because the incrementing / decrementing behavior is tested in the new prose test.

Reply:
I'm also adding test cases for the server.command method in load balanced mode but I don't think that needs to stop a primary review 😄